src/main/java/net/neoforged/camelot/commands/utility/ThreadPingsCommand.java

```diff
@@ -0,0 +1,261 @@
+import com.jagrosh.jdautilities.command.SlashCommandEvent;
+import net.dv8tion.jda.api.entities.IMentionable;
+import net.dv8tion.jda.api.entities.ISnowflake;
+import net.dv8tion.jda.api.entities.Role;
+import net.dv8tion.jda.api.entities.channel.attribute.IThreadContainer;
+import net.dv8tion.jda.api.entities.channel.middleman.GuildChannel;
+import net.dv8tion.jda.api.entities.channel.middleman.StandardGuildChannel;
+import net.dv8tion.jda.api.entities.channel.unions.GuildChannelUnion;
+import net.dv8tion.jda.api.events.GenericEvent;
+import net.dv8tion.jda.api.events.interaction.component.EntitySelectInteractionEvent;
+import net.dv8tion.jda.api.interactions.commands.OptionMapping;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.commands.SlashCommandInteraction;
+import net.dv8tion.jda.api.interactions.commands.build.OptionData;
+import net.dv8tion.jda.api.interactions.components.ActionRow;
+import net.dv8tion.jda.api.interactions.components.selections.EntitySelectMenu;
+import net.dv8tion.jda.api.interactions.components.selections.EntitySelectMenu.SelectTarget;
+import net.dv8tion.jda.api.interactions.components.selections.SelectMenu;
+import net.dv8tion.jda.api.utils.MiscUtil;
+import net.neoforged.camelot.Database;
+import net.neoforged.camelot.db.transactionals.ThreadPingsDAO;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class ThreadPingsCommand extends SlashCommand {
+    private static final String SELECT_MENU_ID_PREFIX = "thread-pings-configure-";
+
+    public ThreadPingsCommand() {
+        this.name = "thread-pings";
+        this.guildOnly = true;
+        this.children = new SlashCommand[]{
+                new ConfigureChannel(),
+                new ConfigureGuild(),
+                new View(),
+        };
+    }
+
+    @Override
+    protected void execute(SlashCommandEvent event) {
+    }
+
+    public static void onEvent(final GenericEvent gevent) {
+        if (!(gevent instanceof EntitySelectInteractionEvent event)) return;
+        assert event.getGuild() != null;
+
+        final String componentId = event.getComponentId();
+        if (!componentId.startsWith(SELECT_MENU_ID_PREFIX)) return;
+
+        final long channelId = MiscUtil.parseSnowflake(componentId.replace(SELECT_MENU_ID_PREFIX, ""));
+        boolean isGuildId = channelId == event.getGuild().getIdLong();
+
+        final GuildChannel channel = event.getJDA().getGuildChannelById(channelId);
+        if (!isGuildId && channel == null) {
+            // TODO: handle this?
+            return;
+        }
+
+        final List<Role> roles = event.getInteraction().getMentions().getRoles();
+        final List<Long> roleIds = roles
+                .stream()
+                .filter(c -> c.getGuild().equals(event.getGuild())) // In case roles from other guilds are included

```

Comment:
I don't think this is possible.

Reply:
I did this based on the `SlashCommandInteractionEvent` javadocs IIRC, which notes that the given roles _may_ be of a different guild.

Reply:
Yeah, you can technically mention roles from different guilds via the `<@&...>` syntax.
Sure, it doesn't really make sense, but it is possible.

Reply:
The Javadoc does indeed mention the fact that they may be from a different guild but those are the javadocs of [`Mentions`](https://docs.jda.wiki/net/dv8tion/jda/api/entities/Mentions.html) which are primarily used for messages, and those can have roles from different guilds. I would assume that this is false for the mentions of a select menu.

However, the Discord [docs](https://discord.com/developers/docs/interactions/message-components#select-menu-object-select-menu-resolved-object) do not confirm nor deny that the roles must be from the same guild. I would be surprised for them to not validate that a user request contains only user roles in the guild (the official client only prompts roles in the guild to begin with) but I guess that check should remain.